src/util/simplify_expr.cpp

```diff
@@ -274,6 +274,98 @@ bool simplify_exprt::simplify_function_application(exprt &expr)
 
     return false;
   }
+  else if(func_id == ID_cprover_string_index_of_func)
+  {
+    const refined_string_exprt &s1 =
+      to_string_expr(function_app.arguments().at(0));
+
+    const auto s1_data_opt = try_get_string_data_array(s1, ns);
+
+    if(!s1_data_opt)
+    {
+      return true;
+    }
+
+    const array_exprt &s1_data = s1_data_opt->get();
+
+    exprt::operandst ops;
+
+    exprt::operandst::const_iterator string_begin;
+    exprt::operandst::const_iterator string_end;
+
+    if(function_app.arguments().at(1).id() == ID_constant)
```

Comment:
Worth clarifying that this case is for index-of-char? Assert that the type of the constant is appropriate?